marklogic-data-hub-central/ui/src/components/advanced-settings/advanced-settings-dialog.tsx

```diff
@@ -179,6 +185,7 @@ const AdvancedSettingsDialog = (props) => {
           setTargetPermissions(response.data.permissions);
           setTargetFormat(response.data.targetFormat);
           setProvGranularity(response.data.provenanceGranularityLevel);
+          response.data.validateEntity === false ? setValidateEntity(defaultValidateEntity):setValidateEntity(response.data.validateEntity) ;
```

Comment:
Can't we get the default value for 'validateEntity' from backend only, instead of UI figuring it out based on its value in the API response. We can set it in UI as well but we need to be sure that its value will be 'false' in the response only the first time when the step is created.
It will be a bit cleaner to just set its default value in backend where the settings object is created for any step. Then UI will only care about the its value in the GET response.

Reply:
@AkshayBajajML You can set default values for the mapping step here: https://github.com/marklogic/marklogic-data-hub/blob/c2e709789a211aea9c16339632f66ffe4361f948/marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/mapping.sjs#L84